### ISSUE TITLE
Do not show warning message on tasks that were not created a long time ago

### ISSUE THREAD

**Issue by gentlementlegen** [ubiquibot/command-start-stop#14]
The bot shows a warning message on tasks that were opened recently, as seen [here](https://github.com/ubiquity/work.ubq.fi/issues/74#issuecomment-2242566257). It should only display this warning above a certain threshold, which comes from the configuration.

A possible cause would be that value missing in the current configuration. If that is the case, the default threshold should be set, and definitely above 0 days.

---

Tasks to be carried out:
- display a warning message when the task is above the configuration threshold
- do not display the warning if it is under that threshold
- change the configuration to accept a string representing a duration instead of a time stamp
- related tests

**Comment by gentlementlegen** [comment]
@Keyrxng would this be the fix for it? https://github.com/ubiquity/work.ubq.fi/issues/74#issuecomment-2242566257

**Comment by Keyrxng** [comment]
yeah @gentlementlegen  https://github.com/ubiquibot/command-start-stop/pull/10/commits/aae3cbac583a433b6c0e52986f462a9546520041 resolved here in #10

**Comment by gentlementlegen** [comment]
Ok cool thank you, I will close this one after we got your fix merged.

**Comment by 0x4007** [comment]
So the font is wrong be sure to match the styles of the old assign message exactly.

**Comment by gentlementlegen** [comment]
I triple checked I cannot see any font difference.
<img width="735" alt="image" src="https://github.com/user-attachments/assets/24f2453c-1e97-4a95-b7f6-54de12efa068">
<img width="735" alt="image" src="https://github.com/user-attachments/assets/b76dbc24-df8e-4004-8098-769159c38efc">
<img width="735" alt="image" src="https://github.com/user-attachments/assets/f5f0db36-202c-4409-8f5a-12fb63b1a102">
<img width="735" alt="image" src="https://github.com/user-attachments/assets/022f4a87-3ae0-4aed-8e49-a2475cdabe90">

Maybe there is a difference on mobile devices if you are using one?

**Comment by Keyrxng** [comment]
https://github.com/ubq-testing/command-start-stop/issues/5#issuecomment-2243850760

![image](https://github.com/user-attachments/assets/108d79e4-d962-4ecb-9ffa-237889f54b29)

**Comment by 0x4007** [comment]
Use <code> not <samp>

![image](https://github.com/user-attachments/assets/f472449f-bd2b-42b6-ae65-f76cac6ff406)

![image](https://github.com/user-attachments/assets/6ccd8e7c-d2cf-4e5b-9184-8df4516fd70f)

**Comment by gentlementlegen** [comment]
@0x4007 Thanks for the screenshot, it seems to be more specific to mobile for the font. Maybe a font fallback happening there.

**Comment by Keyrxng** [comment]
Been reading through the GFM and CommonMark specs, GH docs etc but  haven't been able to find any other workarounds other than `<samp>`. We'll just have to live with the whitespace if using `<code>` blocks so which of the below options is best, included `<samp>` for easy comparison. I can't find any other workarounds with these tags or others such as `<kbd>` etc.

---

### 1

<code>
<table>
<tr><td>Warning!</td><td>This task was created over 10 days ago. Please confirm that this issue specification is accurate before starting.</td></tr><tr><td>Deadline</td><td>Mon, Jul 22, 10:25 PM UTC</td></tr><tr><td>Registered Wallet</td><td>Register your wallet address using the following slash command: `/wallet 0x0000...0000`</td></tr>
</table>
</code>

---

### 2

<code>
<table><tr><td>Warning!</td><td>This task was created over 10 days ago. Please confirm that this issue specification is accurate before starting.</td></tr><tr><td>Deadline</td><td>Mon, Jul 22, 10:25 PM UTC</td></tr><tr><td>Registered Wallet</td><td>Register your wallet address using the following slash command: `/wallet 0x0000...0000`</td></tr></table>
</code>

---

### 3

<samp>
<table><tr><td>Warning!</td><td>This task was created over 10 days ago. Please confirm that this issue specification is accurate before starting.</td></tr><tr><td>Deadline</td><td>Mon, Jul 22, 10:25 PM UTC</td></tr><tr><td>Registered Wallet</td><td>Register your wallet address using the following slash command: `/wallet 0x0000...0000`</td></tr></table>
</samp>

---

**Comment by 0x4007** [comment]
> Been reading through the GFM and CommonMark specs, GH docs etc but haven't been able to find any other workarounds other than `<samp>`. We'll just have to live with the whitespace if using `<code>` blocks so which of the below options is best, included `<samp>` for easy comparison. I can't find any other workarounds with these tags or others such as `<kbd>` etc.

If its only a problem on mobile then perhaps `<samp>` is the best decision!

**Comment by gentlementlegen** [comment]
Number 3 seems to be the best display in your propositions, if that works on mobile then let's go with it.

**Comment by Keyrxng** [comment]
> If its only a problem on mobile then perhaps <samp> is the best decision!

Yeah it's only mobile that `<samp>` seems to have slightly different rendering

> Number 3 seems to be the best display in your propositions, if that works on mobile then let's go with it.

I agree it looks best but no it is somewhat different although personally I think `<samp>` is the way to go because it'll most commonly be seen via desktop vs mobile and I'd rather no whitespace than to accommodate the lesser used platform.


Sounds like option 3 is preferred and that's how the code is right now so no changes needed in this regard

**Comment by gentlementlegen** [comment]
@Keyrxng You mentioned that the bug was fixed, but now the opposite might be happening, see message here: https://github.com/ubiquity/pay.ubq.fi/issues/237#issuecomment-2285326567

I also suggest that we change the configuration format to take a string representing a duration and not a number which is hard to understand and error prone.

**Comment by Keyrxng** [comment]
> @Keyrxng You mentioned that the bug was fixed, but now the opposite might be happening, see message here: [ubiquity/pay.ubq.fi#237 (comment)](https://github.com/ubiquity/pay.ubq.fi/issues/237#issuecomment-2285326567)
> 
> I also suggest that we change the configuration format to take a string representing a duration and not a number which is hard to understand and error prone.

@gentlementlegen you want to add it into the spec here and I'll handle both?

**Comment by gentlementlegen** [comment]
@Keyrxng Done. Also, I guess the font issue should be carried out in a different issue most likely, if there is not once created already.

**Comment by Keyrxng** [comment]
>  Also, I guess the font issue should be carried out in a different issue most likely, if there is not once created already.

I'm certain that we landed on sticking with `<samp>` and supporting the most used platform as it's only an issue on mobile so no need to do anything with that

**Comment by 0x4007** [comment]
I think we need to tweak the qualitative analysis. Somehow I got 0 relevance on my comments which didn't seem to be the case before with gpt3.5 10x samples.

**Comment by gentlementlegen** [comment]
For the `img` it is currently set to 0 so you can just change the configuration. For the relevance I will read the logs more in detail.

**Comment by 0x4007** [comment]
Margin seems imprecise. Perhaps the approach should be reevaluated so that the math is perfect.

**Comment by gentlementlegen** [comment]
@0x4007 It was supposed to be because the length of results is predictable, but when allocating only one token the result seems to be `{}` without the content (should be `{ "1234" : 0.5 }`. I'll investigate.

**Comment by SyedMoin-lab** [comment]
I noticed that the img setting is currently at 0, so we'll just need to tweak that configuration. Regarding the relevance results, I took a closer look at the logs, and it seems like the max_tokens allocated might have been a bit too tight during the dummy test, which is why the relevance came up empty.

To make sure this doesn’t happen again, we could add a small buffer—maybe an extra 10 tokens—to give us some breathing room. Although we expected the result length to be predictable, it turns out that allocating only one token sometimes gives us an empty output (like {} instead of { "1234": 0.5 }). I’ll keep digging into this to find the best fix.
